apps/src/StudioApp.js

```diff
@@ -3000,6 +3004,31 @@ StudioApp.prototype.showRateLimitAlert = function () {
   });
 };
 
+let libraryPreload;
+StudioApp.prototype.loadLibraries_ = function (helperLibraryNames = []) {
+  if (!libraryPreload) {
+    libraryPreload = Promise.all(helperLibraryNames.map(this.loadLibrary_.bind(this)));
+  }
+  return libraryPreload;
+};
+
+StudioApp.prototype.loadLibrary_ = function (name) {
+  if (this.libraries[name]) {
+    return Promise.resolve();
+  }
+
+  return new Promise((resolve, error) => {
+    $.ajax({
+      url: '/libraries/' + name,
+      success: response => {
+        this.libraries[name] = response;
+        resolve();
+      },
+      error,
+    });
+  });
+};
```

Comment:
Did you consider writing these methods with memoize and fetch and async/await? (no promises I got all the below syntax right)
```js
StudioApp.prototype.loadLibraries_ = _.memoize(function (helperLibraryNames = []) {
  return Promise.all(helperLibraryNames.map(name => this.loadLibrary_(name)));
});

StudioApp.prototype.loadLibrary_ = _.memoize(async function (name) {
  const response = await fetch(`/libraries/${name}`)
  this.libraries[name] = await response.text();
});
```

Reply:
Cool, TIL about [`_.memoize`](https://lodash.com/docs/4.15.10#memoize)!  Good call using `fetch` over `$.ajax`, I need to start getting in that habit.

Reply:
+1, TIL about `_.memoize`! Question for either of you: what is the benefit of `fetch` over `$.ajax`?